packages/jest-environment-node/src/index.ts

```diff
@@ -110,6 +116,22 @@ class NodeEnvironment implements JestEnvironment {
     }
     return null;
   }
+
+  compileFunction(code: string, params: Array<string>, filename: string) {
+    if (!compileFunction) {
+      throw new TypeError(
+        "The version of Node you're using does not support compileFunction, please upgrade",
```

Comment:
dunno if we want a more specific error message?

Reply:
I think taking the minimum version from the Nod docs would be helpful here